Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM node:bullseye-slim as build-stage
+WORKDIR /app
+COPY package*.json /app/
+RUN npm install
+COPY ./ /app/
+ARG configuration=production
+RUN npm run build -- --output-path=./dist/out --configuration $configuration
+
+FROM nginx
```

Comment:
Have you checked the default nginx config that this base image includes?
Does this fit for our use?

Reply:
Yes I have. I think there is no change really needed. I think also because only a few (or only one) clients will use the website scalability is not that important for now. 
![grafik](https://github.com/FH-Joanneum-Iron-Road-for-Children/iron-road-for-children-web/assets/35939309/e1418fd2-0879-4b71-9e80-9d98c00900d9)

Reply:
It's not about scalability (scaling would probably anyway not be addressed on the container build level, but k8 "pod/service level"). It's more about the "available-sites/server" config and if it is suited for Single Page Applications (I guess the default config is for static html pages). SPAs need to be configured so that unknown routes respond with the index page as routing is handled internally in the Angular framework and not on a "file path basis".

You can simply test this by starting the container and navigating away from the index page -> copy the link and try to open it in a new tab (or simply try to refresh the page). If you get a 404 the server is not configured correctly (e.g. see config in https://stackoverflow.com/a/49462805 [tryFiles line in location block]).

Another question is the https config - should this be handled by the nginx container or will their be a Ingress controller that already handles encryption and we use http within the cluster (not sure what is planned here, ask @stkiegerl about that).

Reply:
yes you were right, i adapted the nginx.conf and the Dockerfile